packages/flutter/lib/src/services/binding.dart

```diff
@@ -263,30 +264,110 @@ mixin ServicesBinding on BindingBase, SchedulerBinding {
+  /// [SystemChannels.platform] method channel.
+  ///
+  /// By default, returns [ui.AppExitResponse.exit].
+  ///
+  /// {@tool sample}
+  /// This examples shows how an application can cancel (or not) OS requests for
+  /// quitting an application. Currently this is only supported on macOS and
+  /// Linux.
+  ///
+  /// ** See code in examples/api/lib/services/binding/handle_request_app_exit.0.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  /// * [WidgetsBindingObserver.didRequestAppExit], which can be overridden to
+  ///   respond to this message.
+  /// * [WidgetsBinding.handleRequestAppExit] which overrides this method to
+  ///   notify its observers.
+  Future<ui.AppExitResponse> handleRequestAppExit() async {
+    return ui.AppExitResponse.exit;
+  }
+
+  /// Exits the application by calling the native application API method for
+  /// exiting an application cleanly.
+  ///
+  /// This differs from calling `dart:io`'s [exit] function in that it gives the
+  /// engine a chance to clean up resources so that it doesn't crash on exit, so
+  /// calling this is always preferred over calling [exit]. It also optionally
+  /// gives handlers of [handleRequestAppExit] or [AppLifecycleListener]s that
+  /// set an [AppLifecycleListener.onExitRequested] a chance to cancel the
+  /// application exit.
+  ///
+  /// The [exitType] indicates what kind of exit to perform. For
+  /// [ui.AppExitType.cancelable] exits, the application is queried through a
+  /// call to [handleRequestAppExit], where the application can optionally
+  /// cancel the request for exit. If the [exitType] is
+  /// [ui.AppExitType.required], then the application exits immediately without
+  /// querying the application.
+  ///
+  /// For [ui.AppExitType.cancelable] exits, the returned response value is the
+  /// response obtained from the application as to whether the exit was canceled
+  /// or not. Practically, the response will never be [ui.AppExitResponse.exit],
+  /// since the application will have already exited by the time the result
+  /// would have been received.
+  ///
+  /// The optional [exitCode] argument will be used as the application exit code
+  /// on platforms where an exit code is supported. On other platforms it may be
+  /// ignored. It defaults to zero.
+  ///
+  /// See also:
+  /// * [AppLifecycleListener] for a class that can listen to exit requests and
+  ///   respond to them without needing to create a [WidgetsBindingObserver].
+  /// * [WidgetsBindingObserver.didRequestAppExit] for a handler you can
+  ///   override on a [WidgetsBindingObserver] to receive exit requests.
+  @mustCallSuper
+  Future<ui.AppExitResponse> exitApplication(ui.AppExitType exitType, [int exitCode = 0]) async {
+    final Map<String, dynamic>? result = await SystemChannels.platform.invokeMethod<Map<String, dynamic>>(
+      'System.exitApplication',
+      <String, dynamic>{'type': exitType.name, 'exitCode': exitCode},
```

Comment:
Instead of the dreaded `dynamic` here and elsewhere, can these be typed as `Object` or `Object?`?